api/routes_rtfsp.go

```diff
@@ -641,6 +686,50 @@ func (api *API) stopIPFSPrivateNetwork(c *gin.Context) {
 	Respond(c, http.StatusOK, gin.H{
 		"response": gin.H{
 			"network_name": networkName,
+			"state":        "stopped",
+		},
+	})
+}
+
+func (api *API) getIPFSPrivateNetworkStats(c *gin.Context) {
+	username := GetAuthenticatedUserFromContext(c)
+	networkName, exists := c.GetPostForm("network_name")
+	if !exists {
+		FailWithMissingField(c, "network_name")
+		return
+	}
+	logger := api.LogWithUser(username).WithField("network_name", networkName)
+	logger.Info("private ipfs network stats requested")
+	networks, err := api.um.GetPrivateIPFSNetworksForUser(username)
+	if err != nil {
+		api.LogError(err, eh.PrivateNetworkAccessError)(c, http.StatusBadRequest)
+		return
+	}
+	var found bool
+	for _, network := range networks {
+		if network == networkName {
+			found = true
+			break
+		}
+	}
+	if !found {
+		logger.Info("user not authorized to access network")
+		Respond(c, http.StatusUnauthorized, gin.H{
+			"response": "user does not have access to requested network",
+		})
+		return
+	}
+	stats, err := api.orch.NetworkStats(c, &ipfs_orchestrator.NetworkRequest{Network: networkName})
+	if err != nil {
+		api.LogError(err, "failed to get network stats")(c, http.StatusBadRequest)
+		return
+	}
+	Respond(c, http.StatusOK, gin.H{
+		"response": gin.H{
+			"network_name": networkName,
+			"disk_usage":   stats.GetDiskUsage(),
+			"uptime":       stats.GetUptime(),
+			"api":          api.cfg.Orchestrator.Host + stats.GetApi(),
```

Comment:
this doesnt seem right... the host is set by the orchestrator: https://github.com/RTradeLtd/ipfs-orchestrator/blob/master/orchestrator/orchestrator.go#L244

Reply:
also i've already integrated stats reporting here: https://github.com/RTradeLtd/Temporal/blob/V2/api/routes_rtfsp.go#L649

i think we should keep just one of the two

Reply:
Hmm I wonder if it's an error then. Everytime the API would be returned as `:<port-number>` such as `:5432` unless that is intended? We'll keep the stats report you already created.

Reply:
I think you might have forgotten to set the address in the orchestrator config - try generating a new one, its a fairly recent addition

Reply:
I've added a warning to the orchestrator if no address is set: https://github.com/RTradeLtd/ipfs-orchestrator/commit/7e8e0a84ea070ada3ecf6d2bad18adaa4909ee59